### ISSUE TITLE
packages that should be migrated to conda-forge

### ISSUE THREAD

**Issue by bgruening** [bioconda/bioconda-recipes#13964]
A few packages in bioconda, should actually be migrated to conda-forge. This can be a very simple task and is ideally suited to learn both bioconda and conda-forge.

We will keep a list here of packages that we think should be migrated:

- [ ] grep

**Comment by npavlovikj** [comment]
In case this helps, here are few other general packages as well:
- [ ] shapeshifter
- [ ] shapeshifter-cli
- [ ] sharedmem
- [ ] r-ldrtools
- [ ] r-syntactic
- [ ] r-msm
- [ ] r-swamp
- [ ] r-perfmeas
- [ ] r-mess 
- [ ] smhasher
- [ ] fitter
- [ ] gromacs
- [ ] logbook
- [ ] xopen
- [ ] libdeflate
- [ ] optbuild
- [ ] fit_nbinom
- [ ] pytest-workflow
- [ ] cromwell
- [ ] scikits-datasmooth

**Comment by simonvh** [comment]
I'm moving bucketcache and its deps (logbook and represent) to conda-forge. What is the procedure once they're up and running at conda-forge? Delete them from bioconda-recipes via PR?

**Comment by pb-cdunn** [comment]
Somewhat related question:

There is a package in Anaconda on a non-standard channel that I would like to use at build-time for Bioconda packages, but it's not in `conda-forge`. How can I use it in a Bioconda recipe? (At build-time only.)

Would it be easier if that package were actually in conda-forge?

**Comment by dpryan79** [comment]
Regarding what to do once packages are migrated, just put them in the `build-fail-blacklist` file (put the reason why above as a comment). At some point we might due a "grand purge" of such recipes.

**Comment by dpryan79** [comment]
@pb-cdunn Yeah, try to get it into conda-forge. You can probably download the package from anaconda.org and largely copy what's under `info/`.

**Comment by rhpvorderman** [comment]
>  cromwell

I would say that WDL is more aimed at the life sciences than anything else. But I can understand if this needs to be in conda-forge.

If this is the case then I will migrate `pytest-workflow` also to conda-forge as soon as cromwell is moved.

**Comment by dpryan79** [comment]
N.B., recipes can be deleted once migrated to conda-forge.

**Comment by marcelotrevisani** [comment]
Please, remove ``pytest-workflow`` from ``bioconda``

**Comment by jenzopr** [comment]
I reviewed all `r-` packages from above and will comment only *problematic* packages here:

package | comment | suggestion
-------- | ------- | ----------
`r-ic10` | depends on `impute` from bioconductor | 
`r-swamp` | depends on `impute` from bioconductor | 
`r-ontologyplot` | depends on `Rgraphviz` from bioconductor | 
`r-stitch` | not in CRAN | 
`r-sleuth` | not in CRAN | 
`r-rubic` | not in CRAN | 
`r-rtassel` | not in CRAN | 
`r-raceid3` | not in CRAN | 
`r-quorts` | not in CRAN, duplicate? | 
`r-nanopore` | not in CRAN | 
`r-lme4qtl` | not in CRAN | 
`r-gwpcr` | not in CRAN | 
`r-syntactic` | not in CRAN | 
`r-metstat` | APACHE 2.0 license, but no packaged `LICENSE` file | 
`r-popgenreport` | Lists `GPL` as license, but no version |
`r-seqminer | Lists `GPL` as license, but no version |
`r-shazam` | CC BY-SA 4.0 license, but no packaged `LICENSE` file |
`r-tcr` | APACHE 2.0 license, but no packaged `LICENSE` file |
`r-tigger` | CC BY-SA 4.0 license, but no packaged `LICENSE` file |
`r-vcfr` | Lists `GPL` as license, but no version |

Any suggestion how to continue with them would be helpful. Feel free to edit the comment. Thanks!

**Comment by bgruening** [comment]
@jenzopr for packages that depend on bioconductor pkgs, we keep them here in bioconda.

For `APACHE 2.0 license, but no packaged LICENSE file` please drop in a License file in the recipe/folder.

`not in CRAN` let's copy the meta.yaml from bioconda straight to conda-forge.

`Lists GPL as license, but no version` not sure what to do here. Assume GPL2? Can we create upstream isues?

**Comment by dpryan79** [comment]
Actually, for the "not in CRAN" packages, they can stay here :)

**Comment by dpryan79** [comment]
BTW, @jenzopr thanks for the Herculean effort :)

**Comment by dpryan79** [comment]
@pschloss All bioconductor packages will always stay in bioconda. Please ask conda-forge to archive the feedstock.

**Comment by calebzulawski** [comment]
I need `grep` for arm platforms, so I opened a conda-forge/staged-recipes#22313

**Comment by MuellerSeb** [comment]
I also think, that all the perl packages should be migrated/available to/on conda-forge.
There are for example problems like these: https://github.com/conda-forge/lcov-feedstock/pull/10

That could be sorted out by having the perl packages on conda-forge without creating duplicates as criticized in https://github.com/bioconda/bioconda-recipes/issues/2796

**Comment by jvolkening** [comment]
I would just like to confirm, before I do much work on this, that it is still the policy to migrate **all** `perl-*` packages out of Bioconda, regardless of domain. Thanks.

**Comment by eunos-1128** [comment]
The package `rename`, which is written in perl, should be moved to conda-forge.

https://anaconda.org/bioconda/rename
